From 372ec45dcf4c0625739eb417f69cca0eabba9ccf Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:57 -0400
Subject: [fs] btrfs: usage error should not be logged into system log

Message-id: <1407179109-13714-351-git-send-email-sandeen@redhat.com>
Patchwork-id: 87178
O-Subject: [RHEL7.1 PATCH 350/422] btrfs: usage error should not be logged into system log
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Anand Jain <Anand.Jain@oracle.com>

commit 402a0f475938f26b0c7b3175c8163b91a12a8a9a
Author: Anand Jain <Anand.Jain@oracle.com>
Date:   Tue May 20 14:38:11 2014 +0800

    btrfs: usage error should not be logged into system log

    I have an opinion that system logs /var/log/messages are
    valuable info to investigate the real system issues at
    the data center. People handling data center issues
    do spend a lot time and efforts analyzing messages
    files. Having usage error logged into /var/log/messages
    is something we should avoid.

    Signed-off-by: Anand Jain <Anand.Jain@oracle.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/sysfs.c b/fs/btrfs/sysfs.c
index ba2a645..c0dfda5 100644
--- a/fs/btrfs/sysfs.c
+++ b/fs/btrfs/sysfs.c
@@ -374,11 +374,8 @@ static ssize_t btrfs_label_store(struct kobject *kobj,
  struct btrfs_root *root = fs_info->fs_root;
  int ret;
 
- if (len >= BTRFS_LABEL_SIZE) {
-  pr_err("BTRFS: unable to set label with more than %d bytes\n",
-         BTRFS_LABEL_SIZE - 1);
+ if (len >= BTRFS_LABEL_SIZE)
   return -EINVAL;
- }
 
  trans = btrfs_start_transaction(root, 0);
  if (IS_ERR(trans))
-- 
1.7.1